extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/ReactiveMessagingBuildTimeConfig.java

```diff
@@ -0,0 +1,14 @@
+package io.quarkus.smallrye.reactivemessaging.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "reactive-messaging", phase = ConfigPhase.BUILD_TIME)
+public class ReactiveMessagingBuildTimeConfig {
+    /**
+     * Whether or not an health check is published in case the smallrye-health extension is present.
```

Comment:
Is the `in case...` part true? AFAICS, you always drag the `smallrye-health` dependency.

Reply:
It should only depend on the SPI, not on the extension itself.

Reply:
You sure? `smallrye-reactive-messaging-health` doesn't depend on health?

Because if you don't have health, I'm surprised this can work: https://github.com/quarkusio/quarkus/pull/10831/files#diff-0e5b3cb2b027d18a3e49591822e8c947R310 . You would need to use a string for the class name as there's a good chance you wouldn't be able to load the checks if health is not present.

Reply:
smallrye-reactive-messaging-health depends on microprofile health not on the quarkus extension.

Reply:
https://github.com/quarkusio/quarkus/pull/10831/files#diff-0e5b3cb2b027d18a3e49591822e8c947R310 refers to classes from reactive messaging (in a specific module that depends on microprofile health).

Reply:
I just re-tested it does not pull quarkus-health, it's up to the user to use it.

Reply:
We discussed this with @cescoffier on Zulip. Let's merge as is even if not entirely consistent with the other SmallRye extensions.